### PR TITLE
Split parts of rke2.vue into separate pages

### PR DESCRIPTION
**Pull request by eva-vashkevich**
### Summary
Fixes rancher/dashboard#9572
Refactoring of rke2.vue file. Moved version configuration and membership configuration tabs into separate files.

### Occurred changes and/or fixed issues
No behavior should change from this change.

### Areas or cases that should be tested
RKE2 cluster provisioning and editing are affected. NO changes in behavior should be present.

### Areas which could experience regressions
RKE2 cluster provisioning and editing.
[TestPlan.docx](https://github.com/rancher/dashboard/files/12529188/TestPlan.docx)

Test plan is attached.

### CODE REVIEW

**Comment by cnotv** [comment]
Please do not forget to add the issue that the PR is addressed to fix.
```
Fixes #9267
```

File: shell/edit/provisioning.cattle.io.cluster/MemberRolesTab.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<script>
+
+import ClusterMembershipEditor, { canViewClusterMembershipEditor } from '@shell/components/form/Members/ClusterMembershipEditor';
+import CreateEditView from '@shell/mixins/create-edit-view';
+import { Banner } from '@components/Banner';
+import Tab from '@shell/components/Tabbed/Tab';
```
Comment (cnotv) [inline shell/edit/provisioning.cattle.io.cluster/MemberRolesTab.vue:6]: Would it not be more conveniente to keep the tabs in the `rke2.vue` file, so we do not have to import all the time and we may use the component as an abstracted view/page?

Comment (eva-vashkevich) [reply shell/edit/provisioning.cattle.io.cluster/MemberRolesTab.vue:6]: Hi, could you please elaborate on what you mean?

**Comment by cnotv** [comment]
I did a manual QA and I noticed that there's a tiny glitch in the loading of the page on creation.
This happens also on edit.
I am not so sure to be able to identify the reason of this, although it may be that some computation which involves the banner, needs to be handled before. Theoretically the component is not loaded till loading is completed.

https://github.com/rancher/dashboard/assets/5009481/89d18d82-7a58-406a-8e86-3b6f33b7ca60

**Comment by eva-vashkevich** [comment]
@cnotv does it load faster for you on master? I don't seem to see a difference when I switch branches. If it's a pre-existing issue, we might want to track it separately.

**Comment by cnotv** [comment]
> @cnotv does it load faster for you on master? I don't seem to see a difference when I switch branches. If it's a pre-existing issue, we might want to track it separately.

Unfortunately it does not happen on `master` 😞

**Comment by eva-vashkevich** [comment]
@cnotv ohh I see the issue now. Good catch!

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 2]
```diff
@@ -199,18 +201,16 @@ export default {
       set(this.value.spec, 'rkeConfig.machineSelectorConfig', [{ config: {} }]);
     }
 
-    // Store the initial PSP template name, so we can set it back if needed
-    const lastDefaultPodSecurityPolicyTemplateName = this.value.spec.defaultPodSecurityPolicyTemplateName;
-    const previousKubernetesVersion = this.value.spec.kubernetesVersion;
-
     const truncateLimit = this.value.defaultHostnameLengthLimit;
+    const previousKubernetesVersion = this.value.spec.kubernetesVersion;
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:205]: previousKubernetesVersion and lastDefaultPodSecurityPolicyTemplateName have just moved position

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 3]
```diff
@@ -1184,6 +948,9 @@ export default {
         // Store default versions
         this.defaultRke2 = defaultRke2;
         this.defaultK3s = defaultK3s;
+        this.versionsInfo = {
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:951]: This needs to be removed and the specific properties needed by `basics` passed through. that means the computed properties within `basics` can go away and the interface between parent and child containers is cleaner

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 4]
```diff
@@ -1808,16 +1575,6 @@ export default {
       this.syncChartValues(name);
     },
 
-    initQuestions(name) {
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:1811]: comment only: good removal! i'm not going to admit it was me that introduced this and it was never used. nu uh.

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 5]
```diff
@@ -2890,7 +2393,7 @@ export default {
           </Banner>
           <div class="row">
             <div class="col span-6">
-              <h3>Control Plane</h3>
+              <h3>{{ t('cluster.rke2.controlPlaneConcurrency.header') }}</h3>
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:2396]: comment only: nice updates

**Review by richard-cox — CHANGES_REQUESTED** [review]
- This PR moves out the functionality of the basics tab and the visuals of the member roles tab. There's a bit more to do to resolve https://github.com/rancher/dashboard/issues/9572
- There's a loooot of functionality that's touched following these changes. I'm a little nervous about them without a good set of existing automated tests. It would be good to spec out some tests cases that QA should look in to (which parts of the provisioning were you unable to exercise)
  - Update. So far i've tracked..
    - create, edit of clusters from providers Digital Ocean and vSphere (one basic, one advanced)
    - excercise features Basic, Member Roles and Add-On Config tabs
    - PSP functionality
    - The harvester team might need a heads up, just to validate there stuff
- ~There's methods, watches, etc that have moved location in the rke2 component. Those changes make this hard to review. I tracked showCni, hasPsaTemplates, canManageMembers, (watch) showCloudProvider and then stopped.~
- ~Bug - Create DO RKE2 --> `Basics` --> Set Container Network to `cilium`~
   ```
   TypeError: Cannot read properties of undefined (reading 'find')
    at VueComponent.chartVersionKey (Basics.vue:633:1)
    at VueComponent.get (Basics.vue:554:1)
   ```

~Haven't dived in to the Basics component yet, but this should be enough to progress until then.~

~Also, milestone!~

~Edit: probably worth a rebase as well, to remove those two compilation warnings~

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 6]
```diff
@@ -1160,6 +923,7 @@ export default {
 
         const res = await allHash(hash);
 
+        // Everything below this line should be cleaned up
```
[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:926]: Nope, forgot to remove the comment.

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue (hunk 3 above)
Comment (eva-vashkevich) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:951]: why do you think it is cleaner that way? If nothing else uses those properties

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 7]
```diff
@@ -0,0 +1,1006 @@
...
+
+    ciliumIpv6: {
+      get() {
+        // eslint-disable-next-line no-unused-vars
+        const cni = this.serverConfig.cni; // force this property to recalculate if cni was changed away from cilium and chartValues['rke-cilium'] deleted
+
+        return this.userChartValues[this.chartVersionKey('rke2-cilium')]?.cilium?.ipv6?.enabled || false;
+      },
+      set(val) {
+        this.$emit('ciliumIpv6Changed', val);
```
[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply shell/edit/provisioning.cattle.io.cluster/Basics.vue:557]: Doesn't https://github.com/rancher/dashboard/pull/9566/files#diff-6836ab4af70ed1f33db60d2de9a0087b636d7604c3f1cf6bb6525f50f16a69aeR765 handle it?

[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply shell/edit/provisioning.cattle.io.cluster/Basics.vue:557]: Nvm, I know what you meant now

**Comment by richard-cox** [comment]
@eva-vashkevich there's conflicts given the changes I introduced in another PR. just let me know if there's any questions or you'd like me to do it

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 8]
```diff
@@ -410,196 +385,6 @@ export default {
...
-          });
-        }
-      }
-
-      const cur = this.agentConfig['cloud-provider-name'];
-
-      if (cur && !out.find((x) => x.value === cur)) {
-        // Localization missing
-        // Look up cur in the localization file
-        const label = this.$store.getters['i18n/withFallback'](`cluster.cloudProvider."${ cur }".label`, null, cur);
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:562]: this has been dropped from cloudProviderOptions in Basics

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 9]
```diff
@@ -410,196 +385,6 @@ export default {
...
-
-        out.unshift({
-          label:       `${ label } (Current)`,
-          value:       cur,
-          unsupported: true,
-          disabled:    true
-        });
-      }
-
-      const initial = this.initialCloudProvider;
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:572]: this and the if block below have also been dropped

**Review by richard-cox — CHANGES_REQUESTED** [review]
Some comments, and I've updated https://github.com/rancher/dashboard/pull/9566#pullrequestreview-1602931367 with the remaining items

**Comment by eva-vashkevich** [comment]
@richard-cox do you know who I should check with on harvester side?

**Review by richard-cox — APPROVED** [review]
Think @mantis-toboggan-md should give this a once over as well before merging.

The test plans need to request the functionality is exercised as well, but it doesn't need to be specific. In terms of harvester side Nancy can advice, but a heads up in their slack channel about the change should be enough.

I've added the milestone....

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 10]
```diff
@@ -0,0 +1,738 @@
...
+    showDeprecatedPatchVersions: {
+      type:     Boolean,
+      required: true
+    }
+
+  },
+
+  data() {
+    return {
+      clusterIsAlreadyCreated: !!this.value.id,
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:113]: It looks like this is also defined and used in rke2.vue - could it be passed into Basics.vue as a prop?

Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:113]: It looks like this is already defined in rke2.vue - could it be passed into Basics.vue as a prop?

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 11]
```diff
@@ -0,0 +1,738 @@
...
+      initialCloudProvider:    this.value?.agentConfig?.['cloud-provider-name'],
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:114]: This is also defined in rke2.vue

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 12]
```diff
@@ -0,0 +1,738 @@
...
+
+    /**
+     * Check presence of PSPs as template or CLI creation
+     */
+
+    hasPsps() {
+      return !!this.psps?.count;
+    },
+
+    isElementalCluster() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:132]: This is defined in rke2.vue

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 13]
```diff
@@ -0,0 +1,738 @@
...
+    },
+
+    /**
+     * Define introduction of Rancher defined PSA templates
+     */
+    hasPsaTemplates() {
+      return !this.needsPSP;
+    },
+
+    isK3s() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:162]: This is defined in rke2.vue

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 14]
```diff
@@ -0,0 +1,738 @@
...
+      if ( !this.agentArgs['cloud-provider-name'] ) {
+        return false;
+      }
+
+      const name = this.agentConfig['cloud-provider-name'];
+
+      return name === 'rancher-vsphere';
+    },
+
+    showCni() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:408]: This is defined in rke2.vue

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 15]
```diff
@@ -0,0 +1,738 @@
...
+        const cni = this.serverConfig.cni; // force this property to recalculate if cni was changed away from cilium and chartValues['rke-cilium'] deleted
+
+        return this.userChartValues[this.chartVersionKey('rke2-cilium')]?.cilium?.ipv6?.enabled || false;
+      },
+      set(val) {
+        this.$emit('ciliumIpv6Changed', val);
+      }
+    },
+
+    isHarvesterExternalCredential() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:439]: This is defined in rke2.vue. It looks like if you make this a prop, you wont need the `credential` prop

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 16]
```diff
@@ -0,0 +1,738 @@
...
+      return this.credential?.harvestercredentialConfig?.clusterType === 'external';
+    },
+
+    unsupportedCloudProvider() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:443]: This is _not_ defined in rke2.vue, but rke2.vue references it (L1972), so I think it needs to be moved there and passed in as a prop here.

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 17]
```diff
@@ -0,0 +1,738 @@
...
+      const version = release.match(/\d+/g);
+      const isRequiredVersion = version?.length ? +version[0] > 1 || +version[1] >= 23 : false;
+
+      return isRequiredVersion;
+    },
+
+    /**
+     * Define introduction of Rancher defined PSA templates
+     */
+    hasPsaTemplates() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:158]: This is defined in rke2.vue

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 18]
```diff
@@ -263,10 +260,6 @@ export default {
     ...mapGetters({ features: 'features/get' }),
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:260]: This can be removed - looks like `features` is only used in Basics.vue

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

File: shell/edit/provisioning.cattle.io.cluster/Basics.vue [hunk 19]
```diff
@@ -104,22 +103,55 @@ export default {
...
+      required: true
+    },
+    unsupportedCloudProvider: {
+      type:     Boolean,
+      required: true
     }
-
   },
 
   data() {
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/Basics.vue:149]: It looks like `data` can be removed now. PR looks good otherwise

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

**Review by mantis-toboggan-md — APPROVED** [review]

**Pull request merged**